src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -693,8 +694,12 @@ ssize_t sockinfo_tcp::tx(const tx_call_t call_type, const struct iovec* p_iov, c
 		return -1;
 	}
 	si_tcp_logfunc("tx: iov=%p niovs=%d dummy=%d", p_iov, sz_iov, is_dummy);
-	lock_tcp_con();
 
+	if (unlikely(MCE_SPEC_STAC == m_sysvar_mce_spec)) {
```

Comment:
I suggest adding a new boolean configuration parameter named VMA_TCP_RX_POLL_ON_TX (by default is false (0)
Then you should check this parameter
```
if (unlikely(m_sysvar_mce_tcp_rx_poll_on_tcp == true))
```
instead of
```
if (unlikely(MCE_SPEC_STAC == m_sysvar_mce_spec))
```
In VMA_SPEC=stac you should set VMA_TCP_RX_POLL_ON_TX to true (1)

TODO note: we should learn how to RX-poll automatically based on TCP window parameters rather than using VMA_TCP_RX_POLL_ON_TX parameter